src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import classNames from 'classnames';
+
+type Props = {
+  setOpenedTodo: (todo: Todo | null) => void;
+  setShowModal: (showModal: boolean) => void;
+  openedTodo: Todo | null;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  setOpenedTodo,
+  setShowModal,
+  openedTodo,
+}) => {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+
+  useEffect(() => {
+    if (!openedTodo) {
+      return;
+    }
+
+    getUser(openedTodo.userId).then(userInfo => {
+      setUsername(userInfo.name);
+      setEmail(userInfo.email);
+    });
```

Comment:
The `getUser` function is called to fetch user information based on `openedTodo.userId`. Consider adding error handling for the promise returned by `getUser` to manage potential errors in the data fetching process. This will improve the robustness of the component in case of network issues or server errors.